lib/mix/test/mix/tasks/do_test.exs

```diff
@@ -10,16 +10,4 @@ defmodule Mix.Tasks.DoTest do
       assert_received {:mix_shell, :info, ["mix compile.app" <> _]}
     end
   end
-
-  test "gather_command ignore spaces and trailing commas" do
-    import Mix.Tasks.Do, only: [gather_commands: 1]
-    assert gather_commands(["compile", "--list,", "help"]) == [["compile", "--list"], ["help"]]
-    assert gather_commands(["compile", "--list,help"]) == [["compile", "--list"], ["help"]]
-    assert gather_commands(["help", ",compile", "--list"]) == [["help"], ["compile", "--list"]]
-    assert gather_commands(["compile", "--list", ",", "help"]) == [["compile", "--list"], ["help"]]
```

Comment:
I think we should keep the tests but adjust them accordingly. I think it may reveal cases we can handle, like this one.